parquet-hadoop/src/main/java/org/apache/parquet/filter2/statisticslevel/StatisticsFilter.java

```diff
@@ -166,12 +174,17 @@ private boolean hasNulls(ColumnChunkMetaData column) {
       return isAllNulls(meta);
     }
 
-    if (hasNulls(meta)) {
+    if (stats.isNumNullsSet() && hasNulls(meta)) {
```

Comment:
(optional) You might consider adding the "`stats.isNumNullsSet()`" clause to the body of `hasNulls()` itself, but I don't really have a strong opinion on this.

Reply:
It would not work correctly because `hasNulls` is called negated as well.